testcase/cross-region/tso.go

```diff
@@ -0,0 +1,250 @@
+package crossregion
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+
+	util2 "github.com/pingcap/tipocket/testcase/cross-region/pkg/util"
+)
+
+func (c *crossRegionClient) testTSO(ctx context.Context) error {
+	if err := c.requestTSOs(ctx); err != nil {
+		return err
+	}
+	log.Info("start to transfer Leader")
+	if err := c.transferLeader(); err != nil {
+		return err
+	}
+	log.Info("leader transfer committed")
+	if err := c.waitLeaderReady(); err != nil {
+		return err
+	}
+	log.Info("new leader ready")
+	for i := 1; i <= 3; i++ {
+		if err := c.transferPDAllocator(fmt.Sprintf("dc-%d", i)); err != nil {
+			return err
+		}
+	}
+	log.Info("new allocators ready")
+	c.requestTSOAfterTransfer(ctx, "global")
+	c.requestTSOAfterTransfer(ctx, "dc-1")
+	c.requestTSOAfterTransfer(ctx, "dc-2")
+	c.requestTSOAfterTransfer(ctx, "dc-3")
```

Comment:
I think we can mix the global and local requests here like:

dc-1
global
dc-2
global
dc-3